docs/configuration.md

```diff
@@ -1940,6 +1944,59 @@ The TCP address to connect to, including a port number.
   </tr>
 </table>
 
+## Proxy
+
+    proxy:
+      remoteurl: https://registry-1.docker.io
+      username: [username]
+      password: [password]
+
+Proxy enables a registry to be configured as a pull through cache to the official Docker Index.  See [mirror.md](mirror.md) for more information
+
+<table>
+  <tr>
+    <th>Parameter</th>
+    <th>Required</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>
+      <code>remoteurl</code>
+    </td>
+    <td>
+      yes
+    </td>
+    <td>
+     The URL of the official Docker Index
+    </td>
+  </tr>
+  <tr>
+    <td>
+      <code>username</code>
+    </td>
+    <td>
+      no
+    </td>
+    <td>
+     The username of the Docker Index account
+    </td>
+  </tr>
+  <tr>
+    <td>
+      <code>password</code>
+    </td>
+    <td>
+      no
+    </td>
+    <td>
+     The password for the official Docker Index account
+    </td>
+  </tr>
+</table>
+
+To enable pulling private repositories (e.g. `batman/robin`) a username and password for user `batman` must be specified. 
```

Comment:
A word about what that means in term of privacy of the content is probably welcome.

"By allowing your mirror to access your private resources on the Hub, you are delegating the responsibility to secure access to these resources to the mirror itself. Consider yourself warned, cowboy."

Or something like that in a more appropriate language :)